--- modules/pari/desc2lisp.lisp.orig	2021-06-28 14:32:42.000000000 -0600
+++ modules/pari/desc2lisp.lisp	2022-09-19 21:12:06.708459685 -0600
@@ -166,7 +166,15 @@
       (setq h (car h)))
     (if (and h (ext:starts-with-p h f) (setq end (search "):" h)))
         ;;(parse-arglist-regexp h :start (1+ (position #\( h)) :end end)
-        (parse-arglist-read h :start (position #\( h) :end (1+ end))
+	(let ((l (parse-arglist-read h :start (position #\( h) :end (1+ end))))
+	  (and
+	    (consp l)
+	    (consp (cdr l))
+	    (consp (cadr l))
+	    (symbolp (caadr l))
+	    (equalp (symbol-name (caadr l)) "T")
+	    (setf (caadr l) 'T_))
+	  l)
         ;; (or (warn "~S: no arglist in ~S" 'desc-arglist desc) ...)
         :unspecific)))
 
--- modules/pari/pari.lisp.orig	2021-06-28 14:32:42.000000000 -0600
+++ modules/pari/pari.lisp	2022-09-19 21:23:25.026083675 -0600
@@ -218,8 +218,7 @@ t.e., this is the memory size for the re
 ;; extern int factor_add_primes;
 (def-c-var factor_add_primes (:type int))
 
-;; extern ulong DEBUGFILES, DEBUGLEVEL, DEBUGMEM
-(def-c-var debugfiles (:name "DEBUGFILES") (:type ulong))
+;; extern ulong DEBUGLEVEL, DEBUGMEM
 (def-c-var debuglevel (:name "DEBUGLEVEL") (:type ulong))
 (def-c-var debugmem (:name "DEBUGMEM") (:type ulong))
 
