--- src/record.d.orig	2017-02-24 10:42:57.482011489 -0700
+++ src/record.d	2017-02-24 10:52:21.862184109 -0700
@@ -1540,7 +1540,7 @@ global maygc object update_instance (obj
       ptr->inst_class_version = obj;
       clr_break_sem_1(); /* permit interrupts again */
     }
-    ASSERT(Record_flags(STACK_(2+4+2*kept_slots)) & instflags_forwarded_B);
+    ASSERT(record_flags(TheInstance(STACK_(2+4+2*kept_slots))) & instflags_forwarded_B);
     dotimesL(kept_slots,kept_slots, {
       var object new_slotinfo = popSTACK();
       ASSERT(atomp(new_slotinfo));
